flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,442 @@
+
+            if entry.get('file'):
+                logger.info('  File: {}', entry['file'])
+            elif entry.get('url'):
+                logger.info('  Url: {}', entry['url'])
+            else:
+                logger.info('  No URL or File found.')
+
+            logger.info('  Destination: {}', destination)
+            logger.info('  Tags: {}', ', '.join(tags) or 'None')
+            logger.info('  Start: {}', config.get('start', True))
+            return
+
+        if entry.get('file'):
+            if not os.path.exists(entry['file']):
+                return entry.fail('Temp file not found.')
+
+            with open(entry['file'], 'rb') as fs:
+                encoded_file = base64.b64encode(fs.read()).decode('utf-8')
+                if not self.flood.add_torrent_files(
+                    config, files=[encoded_file], destination=destination, tags=tags, start=start
+                ):
+                    return entry.fail('Failed to add file(s) to Flood.')
+        elif entry.get('url'):
+            if not self.flood.add_torrent_urls(
+                config, urls=[entry['url']], destination=destination, tags=tags, start=start
+            ):
+                return entry.fail('Failed to add url(s) to Flood.')
+        else:
+            return entry.fail('No URL or File found.')
+
+    @plugin.priority(120)
+    def on_task_download(self, task: Task, config: dict):
+        """
+        Use the download plugin to get temp files for entry URLs.
+        """
+
+        if config['action'] == 'add' and 'download' not in task.config:
+            download = plugin.get('download', self)
+            download.get_temp_files(task)
+
+    @plugin.priority(135)
+    def on_task_output(self, task: Task, config: dict):
+        if not imported:
+            raise DependencyError('pyjwt', 'pyjwt', 'pyjwt is required for this plugin')
+
+        # Don't do anything if no config provided
+        if not isinstance(config, dict):
+            return
+
+        # Authenticate with Flood if not already authenticated.
+        if self.flood.is_jwt_expired():
+            logger.debug("JWT token is expired. Re-authenticating with Flood.")
+            self.flood.authenticate(config)
+
+        for entry in task.accepted:
+            if config['action'] == 'add':
+                self.add_entry(config, task, entry)
+            elif 'flood_hash' in entry:
+                if config['action'] == 'remove':
+                    self.flood.delete_torrents(config, [entry['flood_hash']], False)
+                elif config['action'] == 'delete':
+                    self.flood.delete_torrents(config, [entry['flood_hash']], True)
+                elif config['action'] == 'start':
+                    self.flood.start_torrents(config, [entry['flood_hash']])
+                elif config['action'] == 'stop':
+                    self.flood.stop_torrents(config, [entry['flood_hash']])
```

Comment:
Consider extracting this to remove the nested if/else.

Reply:
Extracted this into it's own method in winter/Flexget@584f9b7